src/components/PolarisTestProvider/PolarisTestProvider.tsx

```diff
@@ -114,7 +114,7 @@ export function PolarisTestProvider({
 function noop() {}
 
 function createThemeContext(theme: Partial<Theme> = {}): Theme {
```

Comment:
The arguments of PolarisTestProvider should match that of AppProvider.

The `theme` prop should be of type `ThemeConfig`, and this function should be typed as  `function createThemeContext(theme: ThemeConfig = {}): Theme {`

I think this function can be replaced entirely with your new `buildThemeContext` function now that has been extracted to be sharable